streams/src/main/java/org/apache/kafka/streams/processor/internals/ActiveTaskCreator.java

```diff
@@ -278,45 +236,22 @@ private StreamTask createActiveTask(final TaskId taskId,
     }
 
     void closeThreadProducerIfNeeded() {
-        if (threadProducer != null) {
-            try {
-                threadProducer.close();
-            } catch (final RuntimeException e) {
-                throw new StreamsException("Thread producer encounter error trying to close.", e);
-            }
-        }
-    }
-
-    void closeAndRemoveTaskProducerIfNeeded(final TaskId id) {
-        final StreamsProducer taskProducer = taskProducers.remove(id);
-        if (taskProducer != null) {
-            try {
-                taskProducer.close();
-            } catch (final RuntimeException e) {
-                throw new StreamsException("[" + id + "] task producer encounter error trying to close.", e, id);
-            }
+        try {
```

Comment:
We can change the function name removing `IfNeeded` as well, ditto for similar function names.

Reply:
You are right. I didn't do this yet, because I was thinking about moving StreamsProducer out of ActiveTaskCreator, at which point all of these wrapping functions would go.